### DISCUSSION TITLE
Can't `brew install ghc` in userspace

### DISCUSSION THREAD

**Discussion by jacktose** [Homebrew/discussions#154] [Everyday usage]
I use linuxbrew because I only have permissions within my home directory. [`brew install ghc`][ghc] fails, apparently because it can't access `/tmp/`:

```
$ brew install ghc
...
==> ./configure --prefix=/data/shared/lusers/jacktose/.linuxbrew/Cellar/ghc/8.10.1_2/libexec/integer-gmp --with-pic --disable-shared --build=core2-linux-gnu
==> make
==> make install
cp: error writing ‘/tmp/ghc--binary-20201117-8003-1ylfh5/ghc-8.10.1/compiler/stage2/build/libHSghc-8.10.1_p.a’: No space left on device
...
```
[Full log attached.][ghc.txt]

Is this a me problem or a formula problem? Can I work around it somehow?

[ghc]: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ghc.rb
[ghc.txt]: https://github.com/Homebrew/discussions/files/5556675/ghc.txt

**Comment by gromgit** [comment]
```
cp: error writing ‘/tmp/ghc--binary-20201117-8003-1ylfh5/ghc-8.10.1/compiler/stage2/build/libHSghc-8.10.1_p.a’: No space left on device
```
Not a permissions problem, more like a _disk space_ problem. Create a temp directory somewhere else with more space (e.g. `~/tmp`), then:
```
HOMEBREW_TEMP=~/tmp brew install ghc
```
